pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/ControllerInstanceToggleTest.java

```diff
@@ -16,164 +16,123 @@
-          _helixAdmin.getResourceExternalView(HELIX_CLUSTER_NAME, CommonConstants.Helix.BROKER_RESOURCE_INSTANCE);
-      instanceSet = HelixHelper.getOnlineInstanceFromExternalView(resourceExternalView);
-      Assert.assertEquals(instanceSet.size(), i);
-      LOGGER.trace("Current running broker instance: " + instanceSet.size());
+    // Disable broker instances
+    for (String instanceName : _helixAdmin.getInstancesInClusterWithTag(HELIX_CLUSTER_NAME, BROKER_TENANT_NAME)) {
+      sendPostRequest(_requestBuilder.forInstanceState(instanceName), "Disable");
+      checkNumOnlineInstancesFromExternalView(CommonConstants.Helix.BROKER_RESOURCE_INSTANCE, --numEnabledInstances);
     }
 
-    // Enable BrokerInstance
-    i = 0;
-    for (String instanceName : _helixAdmin.getInstancesInClusterWithTag(HELIX_CLUSTER_NAME, "DefaultTenant_BROKER")) {
-      ExternalView resourceExternalView =
-          _helixAdmin.getResourceExternalView(HELIX_CLUSTER_NAME, CommonConstants.Helix.BROKER_RESOURCE_INSTANCE);
-      Set<String> instanceSet = HelixHelper.getOnlineInstanceFromExternalView(resourceExternalView);
-      Assert.assertEquals(instanceSet.size(), i++);
-      sendPostRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forInstanceState(instanceName), "Enable");
-      Thread.sleep(2000);
-      resourceExternalView =
-          _helixAdmin.getResourceExternalView(HELIX_CLUSTER_NAME, CommonConstants.Helix.BROKER_RESOURCE_INSTANCE);
-      instanceSet = HelixHelper.getOnlineInstanceFromExternalView(resourceExternalView);
-      Assert.assertEquals(instanceSet.size(), i);
-      LOGGER.trace("Current running broker instance: " + instanceSet.size());
+    // Enable broker instances
+    numEnabledInstances = 0;
+    for (String instanceName : _helixAdmin.getInstancesInClusterWithTag(HELIX_CLUSTER_NAME, BROKER_TENANT_NAME)) {
+      sendPostRequest(_requestBuilder.forInstanceState(instanceName), "Enable");
+      checkNumOnlineInstancesFromExternalView(CommonConstants.Helix.BROKER_RESOURCE_INSTANCE, ++numEnabledInstances);
     }
+
     // Delete table
-    sendDeleteRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forTableDelete(tableName));
-    Thread.sleep(2000);
+    sendDeleteRequest(_requestBuilder.forTableDelete(RAW_TABLE_NAME));
     Assert.assertEquals(
         _helixAdmin.getResourceIdealState(HELIX_CLUSTER_NAME, CommonConstants.Helix.BROKER_RESOURCE_INSTANCE)
-            .getPartitionSet().size(), 0);
-
+            .getPartitionSet()
+            .size(), 0);
   }
 
-  private void addOneOfflineSegment(String resourceName) {
-    final SegmentMetadata segmentMetadata = new SimpleSegmentMetadata(resourceName);
-    _pinotResourceManager.addSegment(segmentMetadata, "downloadUrl");
+  private void checkNumOnlineInstancesFromExternalView(String resourceName, int expectedNumOnlineInstances)
+      throws InterruptedException {
+    long endTime = System.currentTimeMillis() + 2000L;
```

Comment:
Are we sure that 2s is long enough for the test to pass in a setup like Travis, where CPU resource is limited and out of our control?